storage/innobase/handler/ha_innodb.cc

```diff
@@ -21468,14 +21468,17 @@ update_innodb_redo_log_encrypt(
  		return;
 	}
 
-	if (srv_redo_log_encrypt != REDO_LOG_ENCRYPT_OFF
-	    && srv_redo_log_encrypt != target) {
-		push_warning_printf(
-			thd, Sql_condition::SL_WARNING,
-			ER_WRONG_ARGUMENTS,
+	if (existing_redo_encryption_mode != REDO_LOG_ENCRYPT_OFF
+	    && existing_redo_encryption_mode != target) {
+		static const char err_msg[] = 
 			" Redo log encryption mode"
 			" can't be switched without stopping the server and"
-			" recreating the redo logs.");
+			" recreating the redo logs.";
```

Comment:
I think, it is better to show(and print) the current detected redo log encryption mode (note that even though variable is OFF, detected mode could be MK or Keyring) and the desired mode.

Reply:
I can change it, but this message is only shown when the user wants to set keyring or master key - so obliviously the existing mode is the other.